src/resolve/treeContainers.ts

```diff
@@ -120,66 +120,55 @@ export class NodeFSTreeContainer extends TreeContainer {
   }
 }
 
-interface ZipEntry {
-  path: string;
-  stream?: () => unzipper.Entry;
-  buffer?: () => Promise<Buffer>;
-}
-
 /**
- * A {@link TreeContainer} that utilizes the central directory of a zip file
- * to perform I/O without unzipping it to the disk first.
+ * A {@link TreeContainer} that performs I/O without unzipping it to the disk first.
  */
 export class ZipTreeContainer extends TreeContainer {
-  private tree = new Map<SourcePath, ZipEntry[] | ZipEntry>();
+  private zip: JSZip;
 
-  private constructor(directory: unzipper.CentralDirectory) {
+  private constructor(zip: JSZip) {
```

Comment:
this avoids the declaration on 127 and the assignment on 131.
```suggestion
  private constructor(private zip: JSZip) {
```